lib/yard/parser/source_parser.rb

```diff
@@ -476,9 +476,8 @@ def convert_encoding(content)
           content.force_encoding('binary')
           ENCODING_BYTE_ORDER_MARKS.each do |encoding, bom|
             bom.force_encoding('binary')
-            if content[0, bom.size] == bom
-              content.force_encoding(encoding)
-              return content
+            if content.start_with?(bom)
```

Comment:
Not sure there's a good enough reason to switch to this more restrictive (newer) API given the simplicity of the existing line.

Reply:
I changed it for two reasons, first, `content[0, bom.size]` creates a string, second, both `content` and `bom` are both forced to `binary` encoding, so there won't be any encoding issues with a character based test, which also happens with the existing check.  Not sure why you're calling it 'more restrictive'?

Regardless, I can revert that change if you'd like...

Reply:
I remember start_with? being added in Ruby 2.5+, but it seems like it was added much earlier-- in which case this is fine.

Reply:
> I remember start_with? being added in Ruby 2.5+

My memory isn't good for that sort of thing.  CI did pass...  I remember a `starts_with?` a very long time ago.  Thanks.